nilearn/plotting/tests/test_html_stat_map.py

```diff
@@ -1,33 +1,342 @@
 import warnings
+from io import BytesIO
+import sys
 
-from nose.tools import assert_equal
 import numpy as np
+from numpy.testing import assert_warns, assert_raises
+
+from nibabel import Nifti1Image
 
 from nilearn import datasets, image
 from nilearn.plotting import html_stat_map
+from nilearn.image import new_img_like
+from ..js_plotting_utils import colorscale
 
 
-def test_encode_nii():
-    mni = datasets.load_mni152_template()
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
+def _check_html(html):
+    """ Check the presence of some expected code in the html
+    """
+    assert isinstance(html, html_stat_map.StatMapView)
+    assert "var brain =" in str(html)
+    assert "overlayImg" in str(html)
 
-    mni = image.new_img_like(mni, np.asarray(mni.get_data(), dtype='>f8'))
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
 
-    mni = image.new_img_like(mni, np.asarray(mni.get_data(), dtype='<i4'))
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
+def _assert_warnings_in(set1, set2):
+    """ Check that warnings are inside a list
+    """
```

Comment:
The docstring does not accurately represent what the code is doing. Rephrasing it would be helpful.

Reply:
I've removed that function: it was a one-liner, and used only once.